backend/bamboo/models.py

```diff
@@ -0,0 +1,278 @@
+from datetime import datetime
+
+from sqlalchemy import Column, Integer, String, Text, ForeignKey, \
+    Boolean, DateTime, Float
+from sqlalchemy.orm import relationship
+from werkzeug.security import generate_password_hash, check_password_hash
+
+from bamboo.core.extensions import db
+
+
+class User(db.Model):
+    __tablename__ = 'user'
+    id = Column(Integer, primary_key=True)
+    name = Column(String)
+    username = Column(String)
+    password_hash = Column(String)
+    biology = Column(String)
+    introduction = Column(String)
+    created_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+    active = Column(Boolean)
+    profile_image_id = Column(Integer, ForeignKey('media.id'))
+    role_id = Column(String, ForeignKey('role.id'))
+
+    profile_image = relationship('Media', foreign_keys=[profile_image_id])
+    role = relationship('Role', back_populates='users')
+    blog_authors = relationship('BlogAuthor', back_populates='author', cascade='all, delete-orphan')
+
+    @property
+    def password(self):
+        raise AttributeError('Write-only property!')
+
+    @password.setter
+    def password(self, password):
+        self.password_hash = generate_password_hash(password)
+
+    def validate_password(self, password):
+        return check_password_hash(self.password_hash, password)
+
+
+class Role(db.Model):
+    __tablename__ = 'role'
+    id = Column(Integer, primary_key=True)
+    name = Column(String)
+    permissions = Column(Integer)
+    created_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+
+    users = relationship('User', back_populates='role')
+
+
+class Media(db.Model):
+    __tablename__ = 'media'
+    id = Column(Integer, primary_key=True)
+    path = Column(String)
+    content_type = Column(String)
+    created_at = Column(DateTime, default=datetime.utcnow)
+    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+
+
+class Site(db.Model):
+    __tablename__ = 'site'
+    id = Column(Integer, primary_key=True)
+    config = Column(Text)
```

Comment:
Does sqlite3 support JSON natively? It seems so for the latest versions. And both MySQL and PostgreSQL support JSON. Maybe we can update it.

Reply:
Updated. SQLite 3.9 and later supports the JSON field.